src/components/calcite-checkbox/calcite-checkbox.tsx

```diff
@@ -180,10 +180,6 @@ export class CalciteCheckbox {
     if (!scale.includes(this.scale)) this.scale = "m";
   }
 
-  disconnectedCallback() {
```

Comment:
Is this intentional? Seems like the hidden input will get appended twice if the checkbox is removed and added back.

Reply:
yes this is intentional.  When the component disconnects from the DOM, because the input is a child, it also gets destroyed.

Reply:
We already discussed in Teams, but posting for posterity:

> it also gets destroyed.

It shouldn't be. You can see this in the following example:

```ts
import { Component, Element, h } from "@stencil/core";
@Component({
  tag: "calcite-example",
  shadow: true
})
export class CalciteExample {
  @Element()
  el: HTMLCalciteExampleElement;
  someEl: HTMLElement = null;
  connectedCallback(): void {
    this.someEl = document.createElement("div");
    this.el.appendChild(this.someEl);
  }
}
```

```ts
let example = document.createElement("calcite-example");
setInterval(() => {
  document.body.append(example);
  console.log("current element children", example.children.length);
  example.remove();
}, 1000);
```